Fixes #45 : Vagrant Broken Build and Tests

platinummonkey opened the pull request:
bug fixes to build scripts, and vagrant provisioning

https://github.com/auth0/nginx-jwt/issues/45

Commit:
bug fixes to build scripts, and vagrant provisioning

Commit:
fix broken test

twistedstream commented:
@platinummonkey: Sorry I'm a little confused as to what the original issue was that you were trying to fix vs. what you ended up doing to fix it. Can you explain in a little more detail what you're trying to do here? I see that you had a build issue (#45) but it almost looks like it was the result of an attempt to add synced folders (which is a cool idea).

platinummonkey commented:
Sure thing! There were a few things:
1. The docker curl-bash was no longer correct, and docker was (instead of HTTP_301) was replying in plain text to use the newer simplified url.  ( https://get.docker.com/ubuntu --> https://get.docker.com/ )
2. Also added the vagrant user to the docker group so sudo is not required.
3. Auto-copied the "auto-synced" directory to ~/nginx-jwt - npm likes to create symlinks. VMWare and Virtualbox will both will not let you create symlinks inside a "synced" folder. It's considered a security flaw and they require that you explicitly turn this off, so rather than fight it, embrace it and help the next dev. 
4. It appeared the newest version of docker will auto-magically/annoyingly prepend "library/" to any image name if not specified in the FROM syntax on the Dockerfiles, so I gave the proxy-base-image an organization prefix of "auth0/".
5. I updated the docs, before it said "Ubuntu on Mac OSX (via Vagrant)", when really, anyone with vagrant can do this, not OS specific. It's nice since I don't have node 0.12 on my system and don't feel the need to install it on my system path just to run the tests. VM's and containers are great for this, so I opted for testing changes in this controlled environment.
6. Fixed a couple misleading log messages the previous error message said you could pass a string or table as a claim_spec but the docs said an anonymous function and the code expected a function.
7. Then just added an IDE specific .gitignore files. Personally I go under the assumption, if something is likely to not just "me-specific" add it to the .gitignore so it doesn't end up accidentally in a commit.

Side Note: I have another PR coming after these (so I can rebase, I originally did these all at once, but decided to break them out into individual PR's after looking at the issues page) which focuses on adding custom headers via a header_specs argument. (Looking at fixing #42 and #43 )

platinummonkey commented:
@twistedstream any additional elaboration or work required for this?

platinummonkey commented:
bump

platinummonkey commented:
I'm not sure what's going on over at auth0 preventing responsiveness, but I decided to fork nginx-jwt over here: https://github.com/platinummonkey/nginx-jwt  I'm currently merging in a number of these issue fixes, and will be setting up CI as well.
